cmake/Corrosion.cmake

```diff
@@ -100,7 +100,7 @@ function(_corrosion_set_imported_location_deferred target_name base_property out
         if(output_dir_curr_config)
             set(curr_out_dir "${output_dir_curr_config}")
         elseif(output_directory)
-            set(curr_out_dir "${output_directory}")
+            set(curr_out_dir "${output_directory}/${config_type}")
```

Comment:
I think we would need to check if `output_directory` contains a generator expression, and only when it **does not**, then append `${config_type}` to match the expected behavior (see e.g. https://cmake.org/cmake/help/latest/prop_tgt/RUNTIME_OUTPUT_DIRECTORY.html)